### PR TITLE
Enable clcache usage and use it in appveyor

### PR DESCRIPTION
**Pull request by piponazo**
This PR enables the usage of [clcache](https://github.com/frerich/clcache) to reuse the compilation objects produced by MSVC. This change speeds up the Appveyor jobs in approximately 1 minute (the average times of the jobs go down from 3min to 2min). 

I did not give details about how to [install clcache](https://github.com/frerich/clcache/wiki/Installation) because those instructions can be found in the clcache project.  In the appveyor.yml I chose the nuget approach to install it.

### CODE REVIEW

**Comment by clanmills** [comment]
Thanks for doing this, Luis.  Does this change demand installation of clclache, or is it an option?  We shouldn't require MSVC users to install this to build.

**Comment by piponazo** [comment]
It's just an option, if clcache is not in the system, things just keep
working as before :)

**Comment by clanmills** [comment]
Wonderful.  Thanks for the clarification.

**Review by clanmills — APPROVED** [review]
This looks good.  If find_program discovers clcache.exe, it'll be used.   And it's only used on Release builds.  We might need to add an option to say "Don't use CLCACHE" - however let's see.

File: cmake/compilerFlags.cmake [hunk 1]
```diff
@@ -108,6 +108,19 @@ endif ()
...
+    find_program(CLCACHE name clcache.exe
+        PATHS ENV CLCACHE_PATH
+        PATH_SUFFIXES Scripts clcache-4.1.0
+    )
+    if (CLCACHE)
+        message(STATUS "clcache found in ${CLCACHE}")
+        if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+            message(WARNING "clcache only works for Release builds")
+        else()
+            set(CMAKE_CXX_COMPILER ${CLCACHE})
```
Comment (D4N) [inline cmake/compilerFlags.cmake]: For completeness: we should also set the C compiler, as we have one or two C files in the repo.

**Comment by clanmills** [comment]
Really good work here, guys.  Dan, you're a genius at seeing the downside of things.  I looked at the CMake code and never gave the C compiler a thought.  Well done.

**Pull request merged**